### PR TITLE
JS: Skip .js files with a same-named .ts file next to it

### PR DESCRIPTION
**Pull request by asgerf**
Some projects check in compiled `.js` files next to the `.ts` file that produced it such as [extensions.ts](https://github.com/microsoft/vscode/blob/master/build/lib/extensions.ts) and [extensions.js](https://github.com/microsoft/vscode/blob/master/build/lib/extensions.js) in vscode. We now skip the `.js` file in this case.

### CODE REVIEW

File: javascript/extractor/src/com/semmle/js/extractor/Main.java [hunk 1]
```diff
@@ -134,7 +135,8 @@ public void run(String[] args) {
 
     TypeScriptParser tsParser = extractorState.getTypeScriptParser();
     tsParser.setTypescriptRam(extractorConfig.getTypeScriptRam());
-    if (containsTypeScriptFiles()) {
+    boolean containsTypeScriptFiles = containsTypeScriptFiles();
```
Comment (esbena) [inline javascript/extractor/src/com/semmle/js/extractor/Main.java]: is this change really required?

Comment (asgerf) [reply javascript/extractor/src/com/semmle/js/extractor/Main.java]: Ah, you're right it isn't. Originally I only wanted to pay the I/O cost if there were any TypeScript files, but that doesn't work in QLTest where there is one extractor invocation per `.js` file.

**Comment by asgerf** [comment]
@esbena can I get another approval?
